Soccar [UI]/src/nl/soccar/ui/input/GamePad.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    private void polling(Controller controller) {
+        if (!POLL) {
+            return;
+        }
+        if (!controller.poll()) {
+            POLL = false;
+            return;
+        }
+
+        Component[] components = controller.getComponents();
+        for (Component c : components) {
+            Component.Identifier componentIdentifier = c.getIdentifier();
+            processButtonActions(c, componentIdentifier);
+            processAnalogActions(c, componentIdentifier);
+        }
+        processSteerActions();
+    }
+
+    private void processAnalogActions(Component c, Component.Identifier componentIdentifier) {
+        if (!c.isAnalog()) {
+            return;
+        }
+
+        float axisValue = c.getPollData();
+        int axisValueInPercentage = getAxisValueInPercentage(axisValue);
+        // Checks if the component is the X axis.
+        if (componentIdentifier == Component.Identifier.Axis.X) {
+            xAxisPercentage = axisValueInPercentage;
+            return; // Go to next component.
+        }
+        // Checks if the component is the Y axis.
+        if (componentIdentifier == Component.Identifier.Axis.Y) {
+            return; // Go to next component
+        }
+
+        if (axisValueInPercentage <= ACTION_THROTTLE) {
+            throttleAction = ThrottleAction.ACCELERATE;
+        } else if (axisValueInPercentage >= ACTION_REVERSE) {
+            throttleAction = ThrottleAction.REVERSE;
+        } else if (throttleAction != ThrottleAction.BOOST) {
+            throttleAction = throttleAction.IDLE;
+        }
+    }
+
+    private void processSteerActions() {
+        if (xAxisPercentage < ACTION_STEER_LEFT) {
+            steerAction = SteerAction.STEER_LEFT;
+        } else if (xAxisPercentage > ACRION_STEER_RIGHT) {
+            steerAction = SteerAction.STEER_RIGHT;
+        } else {
+            steerAction = SteerAction.NONE;
+        }
+    }
+
+    private void processButtonActions(Component c, Component.Identifier componentIdentifier) {
+        if (!componentIdentifier.getName().matches("^[0-9]*$")) {
+            return;
+        }
+
+        boolean isPressed = c.getPollData() != 0.0F;
+        processBoostAction(componentIdentifier, isPressed);
+        processBrakeAction(componentIdentifier, isPressed);
+    }
+
+    private void processBoostAction(Component.Identifier componentIdentifier, boolean isPressed) {
+        if (componentIdentifier == Component.Identifier.Button._0 && isPressed) {
+            throttleAction = ThrottleAction.BOOST;
+        } else if (componentIdentifier == Component.Identifier.Button._0 && !isPressed && throttleAction != ThrottleAction.ACCELERATE && throttleAction != ThrottleAction.REVERSE) {
+            throttleAction = ThrottleAction.IDLE;
+        }
+
+    }
+
+    private void processBrakeAction(Component.Identifier componentIdentifier, boolean isPressed) {
+        System.out.println(componentIdentifier.toString());

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS106)